drivers/clock_control/clock_control_sam_pmc.c

```diff
@@ -0,0 +1,158 @@
+/*
+ * Copyright (c) 2023 Gerson Fernando Budke <nandojve@gmail.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT atmel_sam_pmc
+
+#include <stdint.h>
+
+#include <zephyr/arch/cpu.h>
+#include <zephyr/device.h>
+#include <zephyr/devicetree.h>
+#include <zephyr/drivers/clock_control.h>
+#include <zephyr/drivers/clock_control/atmel_sam_pmc.h>
+
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(clock_control, CONFIG_CLOCK_CONTROL_LOG_LEVEL);
+
+struct atmel_sam_clock_control_config {
+	uint32_t base;
```

Comment:
I see you removed the config from the driver, why not move the `soc` part into this driver?

Reply:
Hi @pdgendt ,

I'm doing incremental changes and this will be touched in near future for sure.
Since these changes involve a lot of people it is always better keep as simple as possible.